### PR TITLE
Adding rebinning to new Muon GUI

### LINKED ISSUE #24803: Enable Rebin functionality in new Muon GUI
Currently the Rebin functionality passes the entered string straight through to the algorithm. This needs to be updated so it can be either variable or fixed binning and the groups and pairs should be calculated for both the binned and unbinned data.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This adds in handling of the rebinning in the new Muon GUI so it matches that in the old GUI. The calculations are also updated so that when a rebinning is specified the groups and pairs are calculated for both the binned and unbinned input workspaces.

**Report to:** anthony 

**To test:**

Open the new Muon GUI change the binning and click update all. Check that the results match with the old GUI

Fixes #24803 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/calculate_pair_and_group.py [hunk 1]
```diff
@@ -55,11 +65,20 @@ def _get_pre_processing_params(context, run):
...
-    try:
-        rebin_args = context.gui_variables["Rebin"]
-        pre_process_params["RebinArgs"] = rebin_args
-    except KeyError:
-        pass
+    if rebin:
+        try:
+            if context.gui_variables['RebinType'] == 'Variable' and context.gui_variables["RebinVariable"]:
+                pre_process_params["RebinArgs"] = context.gui_variables["RebinVariable"]
+        except KeyError:
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/calculate_pair_and_group.py]: can I have an example of when this would fail?

Comment (Mantid-Matthew) [reply scripts/Muon/GUI/Common/calculate_pair_and_group.py]: I think `1,0.5,15,0.1,10` should fail

File: scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py [hunk 2]
```diff
@@ -140,4 +148,38 @@ def update_view_from_model(self):
...
+
+    def from_file_checkbox_changed(self):
+        if self._view.group_range_use_last_data.isChecked():
+            if self._model._data.current_runs:
+                self._view.group_range_max.setText(str(round(max(self._model._data.loaded_data(self._model._data.
+                                                                                               current_runs[-1])['OutputWorkspace'][0].
+                                                                 workspace.dataX(0)), 3)))
+
+            self._view.group_range_max.setEnabled(False)
+            if 'GroupRangeMax' in self._model._data.gui_variables:
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_presenter.py]: sorry, but why do we do this?

**Comment by AnthonyLim23** [comment]
for MUSR62260 if I set rebin to variable and enter `1,5,8` the old GUI gives a warning about data being outside the range. The new GUI just produces an output but has an extra data point. 

If you do a fixed or variable rebin with the value empty then you get the same result as None. Could we autopopulate the values to make it clear that is what will happen?

**Comment by AnthonyLim23** [comment]
I think it would be good if we have a rebin warning to give the user a reminder: 

something like:
“-0.0001”: from min(TOF) to max(TOF) among all events in Logarithmic bins of 0.0001
“*0,100,20000”: from 0 rebin in constant size bins of 100 up to 20,000
“2,-0.035,10”: from 10 rebin in Logarithmic bins of 0.035 up to 10
“0,100,10000,200,20000”: from 0 rebin in steps of 100 to 10,000 then steps of 200 to 20,000

**Comment by AnthonyLim23** [comment]
on variable I cannot get decimals to work.

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by AnthonyLim23** [comment]
unrelated build failure (wish)

**Pull request merged**